TestSequencer still fails

```
Running tool: C:\Program Files\Go\bin\go.exe test -timeout 600s -run ^TestSequencer$ github.com/voedger/voedger/pkg/isequencer -count=100 -v

=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
    c:\Workspace\voedger\pkg\isequencer\impl_test.go:55:
        	Error Trace:	c:/Workspace/voedger/pkg/isequencer/impl_test.go:55
        	Error:      	Not equal:
        	            	expected: 0xc8
        	            	actual  : 0x0
        	Test:       	TestSequencer
    c:\Workspace\voedger\pkg\isequencer\testing.go:1679: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
--- FAIL: TestSequencer (0.00s)
    --- FAIL: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
    c:\Workspace\voedger\pkg\isequencer\impl_test.go:55:
        	Error Trace:	c:/Workspace/voedger/pkg/isequencer/impl_test.go:55
        	Error:      	Not equal:
        	            	expected: 0xc8
        	            	actual  : 0x0
        	Test:       	TestSequencer
    c:\Workspace\voedger\pkg\isequencer\testing.go:1679: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
--- FAIL: TestSequencer (0.00s)
    --- FAIL: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
=== RUN   TestSequencer
=== RUN   TestSequencer/basic_flow
--- PASS: TestSequencer (0.00s)
    --- PASS: TestSequencer/basic_flow (0.00s)
FAIL
FAIL	github.com/voedger/voedger/pkg/isequencer	0.164s
```

--- a/pkg/isequencer/impl_test.go
+++ b/pkg/isequencer/impl_test.go
@@ -9,6 +9,7 @@ import (
 	"context"
 	"errors"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/require"
 
@@ -21,10 +22,11 @@ func TestSequencer(t *testing.T) {
 	t.Run("basic flow", func(t *testing.T) {
 		mockedTime := coreutils.MockTime
 		// Given
+		initialNumber := Number(100)
 		storage := NewMockStorage()
 		storage.SetPLog(map[PLogOffset][]SeqValue{
 			PLogOffset(1): {
-				{Key: NumberKey{WSID: 1, SeqID: 1}, Value: 100},
+				{Key: NumberKey{WSID: 1, SeqID: 1}, Value: initialNumber},
 			},
 		})
 
@@ -33,13 +35,15 @@ func TestSequencer(t *testing.T) {
 		})
 
 		seq, cleanup := New(params, storage, mockedTime)
-
+		defer cleanup()
 		// When
 		offset := WaitForStart(t, seq, 1, 1, true)
 		require.Equal(PLogOffset(2), offset)
 
 		// Generate new sequence Numbers 100 times
-		for i := 1; i <= 100; i++ {
+		const countOfNumbers = 100
+
+		for i := 1; i <= countOfNumbers; i++ {
 			num, err := seq.Next(1)
 			require.NoError(err)
 			require.Equal(Number(100+i), num)
@@ -47,11 +51,20 @@ func TestSequencer(t *testing.T) {
 
 		seq.Flush()
 
-		cleanup()
+		ctx, cancelCtx := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancelCtx()
 
-		nums, err := storage.ReadNumbers(1, []SeqID{1})
-		require.NoError(err)
-		require.Equal(Number(200), nums[0])
+		for ctx.Err() == nil {
+			nums, err := storage.ReadNumbers(1, []SeqID{1})
+			require.NoError(err)
+			// check if number in storage is equal to the expected number
+			if initialNumber+countOfNumbers == nums[0] {
+				return
+			}
+			time.Sleep(10 * time.Millisecond)
+		}
+
+		require.Fail("failed to get expected number from storage")
 	})
 }
 

